dataflow/gpu-workers/e2e_test.py

```diff
@@ -0,0 +1,190 @@
+        [
+            "gcloud",
+            "auth",
+            "configure-docker",
+        ]
+    )
+
+
+def test_python_version(image_name: str, configure_docker: None) -> None:
+    # Make sure the local and Docker Python versions are the same.
+    # If this test fails, the following needs updating:
+    # - noxfile_config.py: The Python 'ignored_versions' should only allow the Dockerfile Python version.
+    # - Dockerfile: The `COPY --from=apache/beam` for the worker boot file.
+    # - Docs tutorial: https://cloud.google.com/dataflow/docs/samples/satellite-images-gpus
+    python_version = (
+        subprocess.run(
+            [
+                "docker",
+                "run",
+                "--rm",
+                "-i",
+                "--entrypoint=bash",
+                image_name,
+                "-c",
+                "python --version",
+            ],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.decode("utf-8")
+        .strip()
+    )
+    assert python_version == "Python 3.6.9"
+
+
+def test_apache_beam_version(image_name: str, configure_docker: None) -> None:
+    # Make sure the installed Apache Beam version matches the Apache Beam image
+    # we use to copy the worker boot file.
+    # If this test fails, the following needs updating:
+    # - Dockerfile: The `COPY --from=apache/beam` for the worker boot file.
+    apache_beam_version = (
+        subprocess.run(
+            [
+                "docker",
+                "run",
+                "--rm",
+                "-i",
+                "--entrypoint=bash",
+                image_name,
+                "-c",
+                "pip freeze | egrep '^apache-beam=='",
+            ],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.decode("utf-8")
+        .strip()
+    )
+    assert apache_beam_version == "apache-beam==2.26.0"
+
+
+def test_tensorflow_version(image_name: str, configure_docker: None) -> None:
+    # Make sure the installed Tensorflow version matches the Tensorflow version
+    # in the Dockerfile.
+    # If this test fails, the following needs updating:
+    # - Dockerfile: The `FROM tensorflow/tensorflow` version.
+    tensorflow_version = (
+        subprocess.run(
+            [
+                "docker",
+                "run",
+                "--rm",
+                "-i",
+                "--entrypoint=bash",
+                image_name,
+                "-c",
+                "pip freeze | egrep '^tensorflow(-gpu)?=='",
+            ],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.decode("utf-8")
+        .strip()
+    )
+    assert tensorflow_version == "tensorflow-gpu==2.4.0"
+
+
+def test_end_to_end(bucket_name: str, image_name: str) -> None:
+    # Run the Beam pipeline in Dataflow making sure GPUs are used.
+    gpu_type = "nvidia-tesla-t4"
+    region = "us-central1"
+    worker_zone = "us-central1-a"
```

Comment:
fyi, there are T4 stockouts in us-central1-a. You can switch to us-central1-f if you are seeing test failures.